### DISCUSSION TITLE
Draft post

### DISCUSSION THREAD

**Discussion by loctran016** [vuepress-theme-hope/vuepress-theme-hope#4681] [Ideas]
### Checklist

- [X] I searched existing issues and no one else requests similar feature.
- [X] I think that 25%+ users are positive towards this feature.

### Describe the feature
```md
---
draft: true
---
```
Will only allow this post(md file) only be viewed (or built) in DEV mode only. Even nuxt-content has this feature built-in

**Comment by Mister-Hope** [comment]
I believe this could be done with a simple hook function:

```ts
export default {
  // ...
  onInitialized: (app) => {
    if (app.env.isBuild) {
      app.pages = app.pages.filter(({ frontmatter }) => frontmatter.draft !== true )
    }
  },
};
```

I do not think this is worthy to have such a feature built-in in my theme, or to publish it as a official plugin.

Can you list its support coverage in other documentation tools? If this is widely supported, then implementing it as a core feature might be expected.

**Reply by yinsel** [reply]
这是其他文档工具支持的draft选项来设置开发可见的草稿页面：https://docusaurus.io/zh-CN/docs/api/plugins/@docusaurus/plugin-content-pages#draft

顺便问一下侧边栏的折叠展开能否添加动画，目前感觉会有点生硬，谢谢。
